### PR TITLE
Added basic exception handler.

### PR DESCRIPTION
**Pull request by jeluard**
### Summary:

Added a basic exception handler displaying exceptions details in a popup for both `dev` and `prod` environment.

This servers 2 purposes:
* do not crash app because of Clojure issues in `prod`
* get more details about issues

This can be modified to display more info or to hook with `instabug` or similar.

status: wip

### CODE REVIEW

File: src/status_im/android/core.cljs [hunk 1]
```diff
@@ -169,7 +170,18 @@
                                        :contact-list-modal contact-list-modal)]
                        [component])]])]]))))})))
 
+(def exc (js/require "react-native-exception-handler"))
```
Comment (rasom) [inline src/status_im/android/core.cljs]: The code related to `react-native-exception-handler` should be moved to separate ns and `register-exception-handler` needs to be added to `s.ios.core` ns.

Comment (jeluard) [inline src/status_im/android/core.cljs]: Sure I will do that! Actually I will probably remove `react-native-exception-handler` as it does really little and go bare metal with `react-native` APIs.
Would it make sense to try to extract common ios/android behaviour in a shared namespace?

Comment (rasom) [inline src/status_im/android/core.cljs]: Yes it would but let's do this in separate PR.

**Comment by flexsurfer** [comment]
@jeluard is it implemented only for Android?

**Comment by jeluard** [comment]
@flexsurfer For now yes, wanted some feedback first :) I will finish this this evening.

**Comment by jeluard** [comment]
It's now implemented using `react-native` API for both `ios` and `android`.

File: src/status_im/utils/utils.cljs [hunk 2]
```diff
@@ -95,3 +95,19 @@
 
 (defn hash-tag? [s]
   (= \# (first s)))
+
+(defn default-alert-handler
+  [error fatal?]
+  (show-popup "Error" (.-message error)))
+
+(def dev-mode? js/__DEV__)
```
Comment (rasom) [inline src/status_im/utils/utils.cljs]: let's use `js/goog.DEBUG`

Comment (jeluard) [inline src/status_im/utils/utils.cljs]: Done

**Comment by jeluard** [comment]
Rebase done.
